### PR TITLE
Select spinlock

### PR DESCRIPTION
**Pull request by Fabien-Chouteau**

### CODE REVIEW

**Comment by Fabien-Chouteau** [comment]
@damaki are you ok with this?

File: tests/rp2040_spinlock/alire.toml [hunk 1]
```diff
@@ -12,3 +12,4 @@ atomic = { path='../..' }
 
 [configuration.values]
 atomic.backend = "rp2040_spinlock"
+atomic.RP2040_Spinlock_ID = 21
```
Comment (damaki) [inline tests/rp2040_spinlock/alire.toml:15]: Did you mean 31 here?

**Comment by damaki** [comment]
> @damaki are you ok with this?

Yes! I like the idea of making it configurable. The only potential edge case you might want to be aware of is the unlikely scenario where:
* the user has a multicore project built against bb-runtimes; and
* they have configured `RP2040_Spinlock_ID` to a value other than 31; and
* the code uses both the legacy __sync built-ins and the __atomic built-ins to access the same data; and
* one core performs an access to the data using __sync; and
* the other core performs an access to the same data using __atomic.

In this case there is a potential race condition since the __sync builtin (implemented by bb-runtimes) uses SPINLOCK31 and the __atomic built-in would use a different spinlock, so both cores could potentially access the same data at the same time.

While it's possible that this could happen, I think in practice it's extremely unlikely because the __sync builtins are unlikely to be used (bb-runtimes only implements two of them for internal use), and even then it's unlikely that both __sync and __atomic would be mixed to access the same data from multiple cores. Do you agree that this risk is acceptably low?

**Comment by Fabien-Chouteau** [comment]
> Do you agree that this risk is acceptably low?

Yes the risk is very low. I don't think people are likely to select another spinlock in practice, but I want it to be possible in case there's a conflict in the future with another crate that would use spinlock31...

File: tests/rp2040_spinlock/alire.toml (hunk 1 above)
Comment (Fabien-Chouteau) [reply tests/rp2040_spinlock/alire.toml:15]: No I am using a value different than the default one on purpose.
